lib/district_repository.rb

```diff
@@ -14,47 +12,59 @@ def find_by_name(name)
     pair = districts.detect do |_key, value|
       name.strip.upcase == value.name.upcase
     end
+
     pair && pair[1]
   end
 
-  def find_all_matching(name_fragment)
+  def find_all_districts_matching_name_fragment(name_fragment)
```

Comment:
Better method names are key. Good methods have names that are descriptive active verbs and serve one purpose - or function.